lib/admin.php

```diff
@@ -148,13 +148,15 @@ function wc_admin_register_pages() {
 			)
 		);
 
-		wc_admin_register_page(
-			array(
-				'title'  => __( 'Stock', 'woocommerce-admin' ),
-				'parent' => '/analytics/revenue',
-				'path'   => '/analytics/stock',
-			)
-		);
+		if ( 'yes' === get_option( 'woocommerce_manage_stock' ) ) {
```

Comment:
I don't think it's blocking, but wanted to point it out. When checking/unchecking the stock checkbox, the menu item only appears/disappears on page refresh or when navigating from a non-wc-admin page to a wc-admin page.

So, if they go to _Settings_ to activate stock management and click on _Save_, when hovering _Analytics_, the _Stock_ report will not be visible yet. Not a big deal, but a small inconvenience that might be confusing?
![image](https://user-images.githubusercontent.com/3616980/54650265-eb5c4080-4aad-11e9-8db0-20553ebb5678.png)

I also noticed the report is available through the URL and it displays _Stock_ data even when stock management is disabled. Should we block any kind of navigation to that page completely? Or maybe showing an empty placeholder in case people end up there? Somebody might have the `wc-admin#/analytics/stock` URL in their history/bookmarks and might be confused seeing stock data while it's disabled in the settings.

Reply:
I spent a bit of time toying with this.  One option is to check if the setting is actively being changed here, but I think the best solution is to change where options updated in WC instead.

Probably a minor issue, but opened an issue [here](https://github.com/woocommerce/woocommerce/issues/23087) to see if we can change the hook options are saved on.